playbooks/files/rax-maas/plugins/neutron_ovs_agent_check.py

```diff
@@ -86,17 +87,68 @@ def check_process_statuses(container_or_host_name, container=None):
 
 def check(args):
 
+    NETWORK_ENDPOINT = '{protocol}://{hostname}:9696'.format(
+        protocol=args.protocol, hostname=args.hostname)
+    try:
+        neutron = get_neutron_client(endpoint_url=NETWORK_ENDPOINT)
+
+    # not gathering api status metric here so catch any exception
+    except Exception as e:
+        metric_bool('client_success', False, m_name='maas_neutron')
+        metric('%s_status' % "neutron-openvswitch-agent",
+               'string',
+               '%s cannot reach API' % "neutron-openvswitch-agent",
+               m_name='maas_neutron')
+        status_err_no_exit(str(e), m_name='maas_neutron')
+        return
+    else:
+        metric_bool('client_success', True, m_name='maas_neutron')
+
+    # gather neutron service states
+    if args.host:
+        agents = neutron.list_agents(host=args.host)['agents']
+    elif args.fqdn:
+        agents = neutron.list_agents(host=args.fqdn)['agents']
+    else:
+        agents = neutron.list_agents()['agents']
+
+    try:
+        ovs_agent = next(
+            a for a in agents if 'openvswitch' in a['binary']
+        )
+    except StopIteration:
+        metric_bool('agents_found', False, m_name='maas_neutron')
+        status_err("No host(s) found in the agents list",
+                   m_name='maas_neutron')
+    else:
+        metric_bool('agents_found', True, m_name='maas_neutron')
+
+    # return all the things
+    status_ok(m_name='maas_neutron')
+
+    agent_is_up = "Yes"
+    if ovs_agent['admin_state_up'] and not ovs_agent['alive']:
+            agent_is_up = "No"
+
+    if args.host:
+        name = '%s_status' % ovs_agent['binary']
+    elif args.fqdn:
+        name = '%z_status' % ovs_agent['binary']
```

Comment:
`%z` typo?

The breakout between host and fqdn seem redundant. More on this below.

Reply:
Followed the same (wrong) code here:
https://github.com/rcbops/rpc-maas/blob/master/playbooks/files/rax-maas/plugins/neutron_service_check.py#L74

Changed it to %s